node/src/config.rs

```diff
@@ -0,0 +1,326 @@
+use colored::Colorize;
+use std::{collections::HashMap, sync::Arc, time::Duration};
+
+use crate::{
+    eth::backend::db::{Db, ForkedDatabase},
+    fork::ClientFork,
+    mem,
+    revm::db::CacheDB,
+};
+use ethers::{
+    core::k256::ecdsa::SigningKey,
+    prelude::{Address, Wallet, U256},
+    providers::{Middleware, Provider},
+    signers::{coins_bip39::English, MnemonicBuilder, Signer},
+    utils::{format_ether, hex, WEI_IN_ETHER},
+};
+use foundry_evm::{
+    executor::fork::{BlockchainDb, BlockchainDbMeta, SharedBackend},
+    revm,
+    revm::{BlockEnv, CfgEnv, TxEnv},
+};
+use parking_lot::RwLock;
+
+pub const NODE_PORT: u16 = 8545;
+
+/// Configurations of the EVM node
+#[derive(Debug, Clone)]
+pub struct NodeConfig {
+    /// Chain ID of the EVM chain
+    pub chain_id: u64,
+    /// Default gas limit for all txs
+    pub gas_limit: U256,
+    /// Default gas price for all txs
+    pub gas_price: U256,
+    /// Signer accounts that will be initialised with `genesis_balance` in the genesis block
+    pub genesis_accounts: Vec<Wallet<SigningKey>>,
+    /// Native token balance of every genesis account in the genesis block
+    pub genesis_balance: U256,
+    /// Signer accounts that can sign messages/transactions from the EVM node
+    pub accounts: HashMap<Address, Wallet<SigningKey>>,
+    /// Configured block time for the EVM chain. Use `None` to mine a new block for every tx
+    pub automine: Option<Duration>,
+    /// port to use for the server
+    pub port: u16,
+    /// maximumg number of transactions in a block
+    pub max_transactions: usize,
+    /// don't print anything on startup
+    pub silent: bool,
+    /// url of the rpc server that should be used for any rpc calls
+    pub eth_rpc_url: Option<String>,
+    /// pins the block number for the state fork
+    pub fork_block_number: Option<u64>,
```

Comment:
Once we can change block number on the node after it's been spun up, we'll also want a flag about whether or not changing block number also changes the block used for RPC queries. This is the same as https://github.com/gakonst/foundry/issues/939

I guess a broader phrasing is: Should the set of cheatcodes/settings used to affect EVM state in `forge` always have RPC method analogs in `anvil`? I'd say yes. Not sure how that affects architecture of this, if at all, but figured it was worth calling out